### DISCUSSION TITLE
用pm2将nonebot2和go-cqhttp都放在后台运行出错

### DISCUSSION THREAD

**Discussion by york-yang-me** [nonebot/discussions#144] [问答]
尝试在树莓派armv8架构上通过pm2启动bot进程和go-cqhttp进程，发现go-cqhttp日志中输出反向WebSocket Universal服务器的连接失败的信息：
![image](https://user-images.githubusercontent.com/54138717/168461027-f72ed54e-7ed7-4190-9771-9e1e3becadf8.png)

![image](https://user-images.githubusercontent.com/54138717/168460995-a20d5e5c-8411-4235-b86d-e82df0c8a8ec.png)
bot的配置文件里的端口和go-cqhttp的端口设置信息如下：
![image](https://user-images.githubusercontent.com/54138717/168461125-866018bc-d522-4ec1-80b3-4d7e58ea1bbc.png)
![image](https://user-images.githubusercontent.com/54138717/168461161-1f3f47a4-7af9-4ab8-a4a7-ec14e161e06f.png)
有尝试过不通过pm2启动bot和go-cqhttp, 或者通过pm2只后台运行go-cqhttp，这两种方式均能进行正常通信，无连接失败的信息。但把nonebot2通过```pm2 start "python -m nb_cli run "```或者```pm2 start nb run```启动的话，就会出现连接拒绝的信息。
希望能给点提示或者解决建议

**Comment by yanyongyu** [comment]
你的nb日志中是否有`application startup complete`字样，截图中并未出现，可能是nonebot未能启动完成

**Reply by york-yang-me** [reply]
嗯，check了一下，确实没有启动完成，一直卡在```scheduler started```，之后就不再输出日志了

**Reply by yanyongyu** [reply]
你是否有某个插件startup卡住了

**Reply by york-yang-me** [reply]
只下了一个插件，刚刚去把[nonebot2成分姬](https://github.com/noneplugin/nonebot-plugin-ddcheck)这个插件移除之后，能够启动了。重新安上之后又不行了，初步判断问题是出在成分姬这个插件上，去翻了下源代码，但还是不太理解为什么用pm2挂载进程后就卡住不动了

**Reply by mnixry** [reply]
怎么感觉又得让htmlrender背锅。。
